lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -795,15 +795,22 @@ def setup_intents(customer_id, payment_method_id, options)
           confirm: true,
           mandate_data: {
             customer_acceptance: {
-              type: "online",
-              online: {
-                ip_address: options.dig(:device_data, :ip),
-                user_agent: options.dig(:device_data, :user_agent)
-              }
             }
           }
         }
 
+        if options[:channel] == "api"
+          post[:mandate_data][:customer_acceptance] = { type: "offline" }
+        else
+          post[:mandate_data][:customer_acceptance] = {
+            type: "online",
```

Comment:
what about a renewal payment? Is this going to have a type `online`?

Reply:
This method is only called when doing a store, so it has nothing to do with doing a renewal 👍